EIPS/eip-7718.md

```diff
@@ -0,0 +1,417 @@
+---
+eip: 7718
+title: Portal Wire Protocol a framework for discv5
+description: A generic framework over discv5 which allows new overlay protocols to be built without reimplementing DHT primitives
+author: Kolby ML (@KolbyML)
+discussions-to: https://ethereum-magicians.org/t/eip-7718-portal-wire-protocol-a-framework-for-discv5/20230
+status: Draft
+type: Standards Track
+category: Networking
+created: 2024-06-06
+---
+
+## Abstract
+
+Discv5 is an extensible protocol which allows building new protocols on top of it. We propose a generic framework for building new DHT networks on top of Discv5 referred to as the `Portal Wire Protocol`
+
+## Motivation
+
+Discv5 is implemented in many languages, has built in packet encryption, used by consensus clients, and allows building sub-protocols using it's `TalkRequest` message type.
+
+With the ubiquity of Discv5 in Ethereum and it being implemented in many languages, building new protocols using Discv5 as a base is beneficial as it enables quicker implementation times, stability as Discv5 is already used in production, as well as inheriting improvements and optimizations discovered by Discv5.
+
+When building new DHT sub-protocols on Discv5, new sub-protocols often share the same primitive message types. By defining the framework `Portal Wire Protocol`, new networks can be built without re-implementing the DHT primitives reducing the development time and inheriting the performance optimizations of the base `Portal Wire Protocol` implementation.
+
+## Specification
+
```

Comment:
If you're using the uppercase keywords (SHOULD, MUST, etc.) you should include the paragraph from the template:

https://github.com/ethereum/EIPs/blob/2ba7edfbc305bf825ef17c459ee6c9d04add9bf4/eip-template.md?plain=1#L55

Reply:
I am linking to the `ethereum/portal-network-specs` now, so this language won't be directly in the EIP, if you still think this warning should be included in this document let me know

Reply:
I am linking to the `ethereum/portal-network-specs` now, so this language won't be directly in the EIP, if you still think this warning should be included in this document let me know

Reply:
> I am linking to the `ethereum/portal-network-specs` now, so this language won't be directly in the EIP, if you still think this warning should be included in this document let me know